lib/ts/fetch.ts

```diff
@@ -285,25 +292,29 @@ export async function onUnauthorisedResponse(
             preAPIResult.requestInit
         );
 
-        let removeIdRefreshToken = true;
-        const idRefreshToken = response.headers.get("id-refresh-token");
-        if (idRefreshToken) {
-            await IdRefreshToken.setIdRefreshToken(idRefreshToken, response.status);
-            removeIdRefreshToken = false;
-        }
+        await saveTokensFromHeaders(response);
 
-        if (response.status === AuthHttpRequest.config.sessionExpiredStatusCode) {
-            // there is a case where frontend still has id refresh token, but backend doesn't get it. In this event, session expired error will be thrown and the frontend should remove this token
-            if (removeIdRefreshToken) {
-                await IdRefreshToken.setIdRefreshToken("remove", response.status);
-            }
+        const isUnauthorised = response.status === AuthHttpRequest.config.sessionExpiredStatusCode;
+
+        // There is a case where the FE thinks the session is valid, but backend doesn't get the tokens.
```

Comment:
I'm not sure if we need this section in react-native, but it could be good to protect against an infinite refresh loop in case the headers/cookies are not getting through.

Reply:
It doesn't hurt to keep it as a precaution, plus again I want to avoid too many differences between the SDKs so some things are part of the react native code even though they are not really required